test/runtime/write.c

```diff
@@ -617,6 +617,85 @@ static void usage(const char *program_name)
            p);
 }
 
+static void fs_stress_test() 
+{
+    int num_files = 1000;
+    int fds[num_files];
+    char fd_names[num_files][50];
+    int i = 0;
+
+    struct timespec start;
+    if (clock_gettime(CLOCK_MONOTONIC, &start) < 0) {
+        perror("bulk_write_test: clock_gettime");
+        goto out_fail;
+    }
+
+    /* Creating and writing to new files */
+    for (; i < num_files; i++) {
+        char buf[BUFLEN];
+        ssize_t rv;
+        sprintf(fd_names[i], "fs_stress_test_%d", i);
+        int fd = open(fd_names[i], O_CREAT | O_RDWR, S_IRUSR | S_IWUSR);
+        fds[i] = fd;
+
+        if (fd < 0) {
+            perror("open");
+            exit(EXIT_FAILURE);
+        }
+
+        _READ(buf, BUFLEN);
+
+        _LSEEK(0, SEEK_SET);
+
+        ssize_t len = strlen(str);
+        _WRITE(str, len);
+    }
+
+    /* Time up to first sync() call */
+    struct timespec postwrite;
```

Comment:
Actually I would like the first stats to be calculated after the first sync() call, not before.